framework/src/auxkernels/AuxKernel.C

```diff
@@ -108,6 +111,12 @@ AuxKernel::AuxKernel(const InputParameters & parameters) :
   for (const auto & it : coupled_vars)
     for (const auto & var : it.second)
       _depend_vars.insert(var->name());
+
+  if (_displaced_var)
+  {
+    addMooseVariableDependency(_displaced_var);
```

Comment:
I think the real action of these two lines are resizing necessary arrays in moose variable.